core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -981,7 +981,13 @@ private[spark] object Utils extends Logging {
       return cached
     }
 
-    val indx: Int = hostPort.lastIndexOf(':')
+    val indx: Int =
+      // Interpret hostPort as literal IPv6 address if it contains two ore more colons
```

Comment:
Nit: ore
You might note here that you're checking that you _don't_ have a `[::1]:123` IPv6 address here -- the braces are key.